app/Http/Controllers/AlbumController.php

```diff
@@ -208,6 +211,32 @@ public function setLicense(SetAlbumLicenseRequest $request): void
 		$request->album()->save();
 	}
 
+	/**
+	 * Upload a track for the Album.
+	 *
+	 * @param SetAlbumTrackRequest $request
+	 * @param Tracks               $tracks
+	 *
+	 * @return void
+	 */
+	public function setTrack(SetAlbumTrackRequest $request, Tracks $tracks): void
+	{
+		$tracks->set($request->album()->id, $request->file);
+	}
+
+	/**
+	 * Upload a track for the Album.
+	 *
+	 * @param DeleteTrackRequest $request
+	 * @param Tracks             $tracks
+	 *
+	 * @return void
+	 */
+	public function deleteTrack(DeleteTrackRequest $request, Tracks $tracks): void
+	{
+		$tracks->delete($request->album()->id);
```

Comment:
Same here. If you already have the album, don't use the ID to fetch the album again.